### PR TITLE
Internal `malloc` and various improvements

### PR DESCRIPTION
**Pull request by 0152la**
Since this has quite a bit of changes, including lots of removals, I decided to open the PR as multiple commits, to make reviewing easier.

* Provide a `malloc` and friends compartment library to be used
  internally, overriding the previous interception implementation. We
  use the DDC capability to get the are of memory that is designated as
  heap space. This is done in a new library, `libcomputils.so`, to be
  loaded in a compartment. This means further libraries loaded will also
  use our own internal `malloc` implementation;
* Due to the above, now removed most of the intercept code; leftover 
code is meant to be rechecked and reworked at some point;
* Fix setting improper bounds for compartment DDC capabilities (i.e.,
  ensure the capability encompasses only the memory region designated
  for a compartment), and now set the offset to match the start of the
  compartment heap, to use for internal `malloc`;
* Further improve symbol relocation lookup;
* Added a few more tests.

### CODE REVIEW

**Comment by ltratt** [comment]
Somehow I missed this one -- sorry!

File: tests/simple_fopen.c [hunk 1]
```diff
@@ -8,31 +8,43 @@ void
...
+    {
+        err(1, "Error in fopen: ");
+    }
     fclose(fd);
 }
 
 void
 by_syscall()
 {
+    int fd = syscall(5, "tmp", O_CREAT); // open
```
Comment (ltratt) [inline tests/simple_fopen.c:21]: Are there named constants for syscalls?

**Comment by ltratt** [comment]
https://github.com/capablevms/CHERI-ELF-comp/pull/27/commits/bd730889c5f622ccedaf4ee0f2625732405dfc21 says "Remove some currently not working tests" but it adds two tests and removes none?

**Comment by 0152la** [comment]
> [bd73088](https://github.com/capablevms/CHERI-ELF-comp/commit/bd730889c5f622ccedaf4ee0f2625732405dfc21) says "Remove some currently not working tests" but it adds two tests and removes none?

I think I forgot to add `CMakeLists.txt` in that commit, and that got added in [06b708a](https://github.com/capablevms/CHERI-ELF-comp/pull/27/commits/06b708a140aaff37405b9025937b6cc654103ef4).

File: tests/simple_fopen.c (hunk 1 above)
Comment (0152la) [reply tests/simple_fopen.c]: Yup; fixed in d91f5e82634dbb57346d78c2588be34059199e04.

**Comment by ltratt** [comment]
Please squash.

**Comment by 0152la** [comment]
Squashed.

**Comment by 0152la** [comment]
I wanted to do a bit further testing before going forward with this PR, in case anything else slipped through the cracks. I'll either squash it, or push further commits, based on how it goes.

**Comment by 0152la** [comment]
Ready for review.

**Comment by 0152la** [comment]
Squashed.

**Pull request merged**